content/en/docs/Tracing/tempo-walkthrough.md

```diff
@@ -0,0 +1,206 @@
+data:
+  tempo-datasource.yaml: |-
+    apiVersion: 1
+    datasources:
+    - name: Tempo
+      type: tempo
+      access: proxy
+      url: "http://tempo.monitoring:3100"
+      version: 1
+      isDefault: true
+EOF
+```
+
+At this point we have a running cluster with Grafana and Tempo backend installed and we can access Grafana using an ingress controller.
+
+## Kyverno Setup
+
+We now need to install Kyverno with tracing enabled and pointing to our Tempo backend.
+
+We can deploy Kyverno using Helm with the following command:
+
+```shell
+helm upgrade --install kyverno --namespace kyverno --create-namespace --wait \
+  --devel --repo https://kyverno.github.io/kyverno kyverno \
+  --values - <<EOF
+# kyverno controller
+extraArgs:
+  # enable tracing
+  - --enableTracing
+  # tempo backend url
+  - --tracingAddress=tempo.monitoring
+  # tempo backend port for opentelemetry traces
+  - --tracingPort=4317
+
+# cleanup controller
+cleanupController:
+  tracing:
+    # enable tracing
+    enabled: true
+    # tempo backend url
+    address: tempo.monitoring
+    # tempo backend port for opentelemetry traces
+    port: 4317
+EOF
+```
+
+## Kyverno policies Setup
+
+Finally we need to deploy a couple of policies in the cluster so that Kyverno can configure admission webhooks accordingly.
+
+We are going to deploy the `kyverno-policies` helm chart (with the `Baseline` profile of PSS) using the following command:
+
+```shell
+helm upgrade --install kyverno-policies --namespace kyverno --create-namespace --wait \
+  --devel --repo https://kyverno.github.io/kyverno kyverno-policies \
+  --values - <<EOF
+validationFailureAction: Enforce
+EOF
+```
+
+Note that we are setting `validationFailureAction` to `Enforce`, this is because `Audit` policies are processed asynchronously and will produce a separate trace from the main one (both traces are linked together though, but not with a parent/child relationship).
+
+## Create a Pod and observe the corresponding trace
+
+With everything in place we can exercise the Kyverno admission webhooks by creating a `Pod` and find the corresponding trace in Grafana.
+
+Run the following command to create a `Pod`:
+
+```shell
+kubectl run nginx --image=nginx
+```
+
+After that, navigate to the [Grafana explore page](http://localhost/grafana/explore), select `Tempo` in the top left drop down list, click on the `Search` tab, and search for traces with the following criterias:
+- Service name: `kyverno`, every trace define a service name and all traces coming from Kyverno will use the `kyverno` service name
+- Span name: `ADMISSION POST /validate/fail`, every span define a span name and root spans created by Kyverno when receiving an admission request have their name computed from the http operation and path (`ADMISSION <HTTP OPERATION> <HTTP PATH>`, the validate `/validate/fail` path indicates indicates that it's a validating webhook that was configured to fail the admission request in case of error).
```

Comment:
"...every span defines a span..."

run-on sentence. Period instead of comma before the clause "the validate..."

Also state that the fail mode here is the default.